### ISSUE TITLE
`-a` option of `borg prune` does not seem to accept `sh:` as pattern selector

### ISSUE THREAD

**Issue by SebastianJL** [borgbackup/borg#7868]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

ISSUE

#### Your borg version (borg -V)
borg 1.2.4 installed via debian apt.
borg 1.2.6 installed via https://github.com/borgbackup/borg/releases/download/1.2.6/borg-linux64

#### Operating system (distribution) and version.
Linux phoenix 6.1.0-10-amd64 <span>#<span>1 SMP PREEMPT_DYNAMIC Debian 6.1.38-1 (2023-07-14) x86_64 GNU/Linux
KDE Plasma Version: 5.27.5

#### Hardware / network configuration, and filesystems used.
Local, but I also observed it on remote with borgbase.com.
Filesystem: ext4.

#### How much data is handled by borg?
```shell
$ du -hs rust_test/
172M    rust_test/
```

#### Full borg command-line that lead to the problem (leave away excludes and passwords)
```
$ borg prune --list -a sh:Day* --keep-last 1 ./borg-repo/
```

## Describe the problem you're observing.
The `-a` option of `borg prune` does not seem to accept `sh:` as pattern selector. From the [documentation](https://borgbackup.readthedocs.io/en/stable/usage/general.html#common-options) it is not entirely clear if this is the intended behavior or a bug.

> -a GLOB, --glob-archives GLOB | only consider archive names matching the glob. **sh: rules apply, see “borg help patterns**”.

Leaving the `sh:` off does the correct thing. Problem here is that `Vorta` produces the pattern and before we know if that needs to be changed we need to know what the intended behavior is and maybe document it better.

Something along the lines of

> -a GLOB, --glob-archives GLOB | only consider archive names matching the glob. sh: rules apply, but you need to leave of the `sh:` selector, see “borg help patterns” for the sh:rules.

See [issue in vorta repo](https://github.com/borgbase/vorta/issues/1829).

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

I ran this with both system wide borg installed via debian apt, and also via the borg 1.2.6 binary just copied into the same directory and calling it via `./borg`

```shell
$ ls -l
total 4
drwxr-xr-x 8 <redacted> <redacted> 4096 Jul  5 15:44 rust_test
$ borg init --encryption=none ./borg-repo
$ borg create ./borg-repo::Day01 rust_test
$ borg create ./borg-repo::Day02 rust_test
$ borg list ./borg-repo
Day01                                Wed, 2023-10-11 10:14:27 [fc3f26653bc067c70e6c00ee2bc9434371796c6363b005532159fdba0f345b23]
Day02                                Wed, 2023-10-11 10:14:43 [ee79208da5f953aeb6acea55357054d157b3c95b9046c6d892e34b0a40354f97]
$ borg prune --list -a sh:Day* --keep-last 1 ./borg-repo/
$ borg list ./borg-repo
Day01                                Wed, 2023-10-11 10:51:01 [13e416e9505c1103115e5b72044d4ebf8103570d0139ad5347a535a8b5aed8d9]
Day02                                Wed, 2023-10-11 10:51:15 [f659f7c0baed878dcc22d8f6a088a7ad0c264ac7238eacf4935a6464b6a6cad6]
$ borg prune --list -a Day* --keep-last 1 ./borg-repo/
Keeping archive (rule: secondly #1):     Day02                                Wed, 2023-10-11 10:14:43 [ee79208da5f953aeb6acea55357054d157b3c95b9046c6d892e34b0a40354f97]
Pruning archive (1/1):                   Day01                                Wed, 2023-10-11 10:14:27 [fc3f26653bc067c70e6c00ee2bc9434371796c6363b005532159fdba0f345b23]
$ borg list ./borg-repo/
Day02                                Wed, 2023-10-11 10:14:43 [ee79208da5f953aeb6acea55357054d157b3c95b9046c6d892e34b0a40354f97]
```

**Comment by SebastianJL** [comment]
So the documentation should maybe even mention that this behavior changes in borg2?

**Comment by SebastianJL** [comment]
FYI: Vorta changed the command generation to reflect borgs behavior in v0.8.11, but I still think it would make sense to add a little something to the borg 1.* documentation.

**Comment by ThomasWaldmann** [comment]
There is no bug in borg here.

borg < 2 always required the archive name pattern to be just a glob (shell) pattern (that's why the option is called `--glob-archives` there), but it never supported giving pattern styles for this (like giving `sh:` or `re:` or ... in front of the pattern).

borg 2 makes this more flexible by supporting pattern styles (which is the reason why the option is called `--match-archives` there).

Yes, the borg 1 docs could be a bit more clear, PRs welcome!

**Comment by SebastianJL** [comment]
@ThomasWaldmann Can you give me a little help. On which branch should I base my change?

**Comment by ThomasWaldmann** [comment]
@SebastianJL The 1.2.x releases are made from branch `1.2-maint`.

`master` branch is what will become borg 2 some day.

**Comment by SebastianJL** [comment]
Thanks for the hint. I see that the .rst files are auto-generated. Can you tell me where they are auto-generated from?

**Comment by ThomasWaldmann** [comment]
Fixed by #7871 in 1.2-maint.
